### PR TITLE
The Share entity has been refactored into a more generic Dataset Logb…

### PR DESCRIPTION
**Pull request by pinarpink**
…ook entity

### CODE REVIEW

**Review by vildead — CHANGES_REQUESTED** [review]
UI testing:
If I choose a partner on Logbook entry, a contract is automatically pre-filled if any exists for given project

When creating logbook entry and choosing value from data log type drop-down list, sometimes the window closes unexpectedly.

File: core/models/data_log_type.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    """
+
+    class Meta:
+        app_label = 'core'
+        get_latest_by = "added"
+        ordering = ['added']
+
+    name = TextFieldWithInputWidget(max_length=128,
+                                    blank=False,
+                                    verbose_name='Name of the type of the document')
```
[the remark this answers is not in the document]
Comment (pinarpink) [reply core/models/data_log_type.py:16]: yep good idea

File: core/models/share.py [hunk 2]
```diff
@@ -5,37 +5,38 @@
 
 class Share(CoreModel):
```
[the remark this answers is not in the document]
Comment (pinarpink) [reply core/models/share.py:6]: there are many things that need cleaning up in the DAISY model. If I rename the model it creates one big hairy mess. I think we can create a ticket and do it for DAISY 2.0

File: core/models/share.py [hunk 3]
```diff
@@ -5,37 +5,38 @@
...
+                                                blank=False,
+                                               on_delete=models.SET_NULL)
+
     partner = models.ForeignKey('core.Partner',
-                                verbose_name='Share partner',
+                                verbose_name='Involved partner',
                                 related_name='shares',
                                 null=True,
                                 on_delete=models.SET_NULL,
                                 blank=False,
```
[the remark this answers is not in the document]
Comment (pinarpink) [reply core/models/share.py:27]: I thought we could enter LCSB as the partner for activities that we do.

[the remark this answers is not in the document]
Comment (pinarpink) [reply core/models/share.py:27]: I am not sure we will have activities that involve more than one partner. Can you give an example?  If so we can enter as several logs.

File: core/models/share.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vildead) [reply core/models/share.py:6]: ok. issue created

File: core/models/share.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vildead) [reply core/models/share.py]: You are absolutely right. Several logs will be enough in case (I cannot thing of now :)

**Review by vildead — APPROVED** [review]
I have added data log types to the admin page.
Just few comments:
- all legacy shares will have the log category unset - is it ok or should we populated the field with 'Transfer'? (fairly easy to do using data migrations)
- when choosing datalog category, the windows sometimes disappears. Nothing in logs :/ 
- testing the form, I somehow got following error but I cannot reproduce it now :/
```
ERROR 2021-05-25 23:40:14,069 log 12748 140559660975936 Internal Server Error: /dataset/3/share/add/
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/daisy/daisy/web/views/share.py", line 58, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/edit.py", line 168, in get
    return super().get(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/edit.py", line 133, in get
    return self.render_to_response(self.get_context_data())
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/edit.py", line 66, in get_context_data
    kwargs['form'] = self.get_form()
  File "/home/daisy/daisy/web/views/share.py", line 73, in get_form
    partner = get_object_or_404(Partner, pk=int(partner))
ValueError: invalid literal for int() with base 10: ' '
```

**Pull request merged**